src/ofp_timer.c

```diff
@@ -481,7 +481,8 @@ int ofp_timer_cancel(odp_timer_t tim)
 		if (timeout_event != ODP_EVENT_INVALID) {
 			tmo = odp_timeout_from_event(timeout_event);
 			bufdata = odp_timeout_user_ptr(tmo);
-			odp_buffer_free(bufdata->buf);
+			if (bufdata != NULL)
+				odp_buffer_free(bufdata->buf);
```

Comment:
Have you seen some problems with this? This user pointer should probably always be non-NULL, so if there's a problem somewhere, this condition just hides the problem, but it doesn't fix it.

Reply:
Yes I had a crash here. I use my own ODP implementation but I can't see how that affects the issue, the timeout user pointer is under the control of the application?
I can debug this further and see the life cycle of the affected timeout.

Reply:
This change is not needed. I had a bug in my clean room ODP implementation, the documentation for odp_timer_set_abs()  isn't very clear on how fields are copied from a timer to its timeout event.